validation/checks/license_check.go

```diff
@@ -0,0 +1,84 @@
+package checks
+
+import (
+	"io"
+	"net/http"
+	"regexp"
+
+        "github.com/UCLALibrary/validation-service/validation/config"
+        "github.com/UCLALibrary/validation-service/validation/csv"
+)
+
+// Error messages
+var (
+	noProfileErr = "supplied profile cannot be nil"
+	urlFormatErr = "license URL is not in a proper format (check for HTTPS)"
+	urlConnectErr = "problem connecting to license URL"
+	urlReadErr = "problem reading body of license URL"
+)
+
+type LicenseCheck struct{
+	profiles *config.Profiles
+}
+
+func (check *LicenseCheck) NewLicenseLCheck(profiles *config.Profiles) (*LicenseCheck, error) {
+	if profiles == nil {
+		return nil, csv.NewError(noProfileErr, csv.Location{}, "nil")
+	}
+
+	return &LicenseCheck{
+		profiles: profiles,
+	}, nil
+}
+
+func (check *LicenseCheck) Validate(profile string, location csv.Location, csvData [][]string) error {
+	// license not relevant to Bucketeer processing
+	if profile == "bucketeer" {
+		return nil
+	}
+
+	if err := csv.IsValidLocation(location, csvData, profile); err != nil {
+		return err
+	}
+
+	// find the header and determine if it matches an ark header
+	header, err := csv.GetHeader(location, csvData, profile)
+
+	if err != nil {
+		return err
+	}
+
+	if header != "License" {
+		return nil
+	}
+
+	value := csvData[location.RowIndex][location.ColIndex]
+
+	if err := verifyLicense(value, profile, location); err != nil {
+		return err
+
+	}
+
+	return nil
+}
+
+func verifyLicense(license string, profile string, location csv.Location) error {
+        r := regexp.MustCompile(`^^http\:\/\/[0-9a-zA-Z]([-.\w]*[0-9a-zA-Z])*(:(0-9)*)*(\/?)([a-zA-Z0-9\-\.\?\,\'\/\\\+&amp;%\$#_]*)?$`)
+        if !r.MatchString(license) {
+                return csv.NewError(urlFormatErr, location, profile)
+        }
+
+        resp, err := http.Get(license)
```

Comment:
Might be nice to keep a map/set of licenses once they're verified (so we don't have to look them up again), but that can be another ticket. I wonder too whether there are ever CSVs with multiple licenses or whether we always have the same license for every item in a CSV. These are just "I wonder" comments, though, not blockers for this ticket.